packages/loot-core/src/types/prefs.d.ts

```diff
@@ -1,5 +1,6 @@
 export type FeatureFlag =
   | 'goalTemplatesEnabled'
+  | 'goalTemplatesUIEnabled'
```

Comment:
Calling this `goalTemplatesUIEnabled` rather than `budgetAutomationsUIEnabled` so the flags are consistent (and a search in the codebase for `goalTemplates` in the future will find all usages)